powerdns/models/powerdns.py

```diff
@@ -278,7 +278,7 @@ def owners(self):
     @property
     def service_owners(self):
         """Returns owners by service unless direct_owners are set"""
-        owners = self.direct_owners.through.objects
+        owners = self.direct_owners.through.objects.filter(domain=self)
```

Comment:
`related_name` is the name from the other side of the relation. it should work, but i'll have to check in shell for exact name